### PR TITLE
Improvement/git add

### LINKED ISSUE #429: Improve Git add functionnality
## Description

As today, we face some issues during the `git add` functionnality. The git add is quite slow because it refreshes all the fileExplorer.
So, we want to update **only** the file added by the previous command instead of recharging everything.

Caution: this ticket is only about `git add`

### PR DESCRIPTION
**Pull request by k-y0u**
resolves #429

### CODE REVIEW

File: tests/unit/components/FileExplorer.spec.js [hunk 1]
```diff
@@ -442,9 +443,9 @@ describe('Test component: FileExplorer', () => {
         path: 'fileName',
         status: 'status',
       }];
-      Project.getStatus.mockImplementation(() => [{ path: 'fileName', status: 'newStatus' }]);
+      // Project.getStatus.mockImplementation(() => [{ path: 'fileName', status: 'newStatus' }]);
```
Comment (justinehell) [inline tests/unit/components/FileExplorer.spec.js:446]: delete comment

File: tests/unit/components/FileExplorer.spec.js [hunk 2]
```diff
@@ -454,9 +455,9 @@ describe('Test component: FileExplorer', () => {
         path: 'fileName',
         status: 'status',
       }];
-      Project.getStatus.mockImplementation(() => [{ path: 'fileName', status: 'status' }]);
+      // Project.getStatus.mockImplementation(() => [{ path: 'fileName', status: 'status' }]);
```
Comment (justinehell) [inline tests/unit/components/FileExplorer.spec.js:458]: delete comment

File: src/components/menu/FileExplorerActionMenu.vue [hunk 3]
```diff
@@ -147,16 +147,23 @@ function deleteFile() {
...
       });
-      GitEvent.AddEvent.next(file.id);
+
+      const [fileStatus] = await getStatus(
+        props.projectName,
+        [file.id],
+        (path) => path === file.id,
+      );
+
+      GitEvent.AddEvent.next(fileStatus);
```
Comment (justinehell) [inline src/components/menu/FileExplorerActionMenu.vue:166]: Update doc for `AddEvent` inside `GitEvent.js`

File: src/components/editor/MonacoEditor.vue [hunk 4]
```diff
@@ -60,7 +61,12 @@ async function updateFile() {
   };
 
   await writeProjectFile(props.projectName, file);
-  FileEvent.UpdateEditorContentEvent.next(file.id);
+  const [fileStatus] = await getStatus(
+    props.projectName,
+    [file.id],
+    (path) => path === file.id,
+  );
+  FileEvent.UpdateEditorContentEvent.next(fileStatus);
```
Comment (justinehell) [inline src/components/editor/MonacoEditor.vue:70]: Update doc for `UpdateEditorContentEvent` inside `FileEvent.js`

**Review by justinehell — CHANGES_REQUESTED** [review]

File: tests/unit/components/FileExplorer.spec.js (hunk 1 above)
Comment (k-y0u) [reply tests/unit/components/FileExplorer.spec.js]: Done

File: src/components/menu/FileExplorerActionMenu.vue (hunk 3 above)
Comment (k-y0u) [reply src/components/menu/FileExplorerActionMenu.vue:166]: Done

File: src/components/editor/MonacoEditor.vue (hunk 4 above)
Comment (k-y0u) [reply src/components/editor/MonacoEditor.vue:68]: Done

Comment (justinehell) [reply src/components/editor/MonacoEditor.vue:68]: you updated `UpdateFileContentEvent` instead of `UpdateFileContentEvent`

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/components/editor/MonacoEditor.vue (hunk 4 above)
Comment (k-y0u) [reply src/components/editor/MonacoEditor.vue:67]: done

**Review by justinehell — DISMISSED** [review]

File: src/components/editor/MonacoEditor.vue [hunk 5]
```diff
@@ -110,7 +111,17 @@ async function createEditor() {
 
   editor = monaco.editor.create(container.value, { value, language });
 
-  editor.onDidChangeModelContent(updateFile);
+  editor.onDidChangeModelContent(() => {
```
Comment (aboubakar14) [inline src/components/editor/MonacoEditor.vue:113]: Maybe add a comment on why we are doing this ?

Comment (k-y0u) [reply src/components/editor/MonacoEditor.vue:113]: changelog

File: src/composables/FileExplorer.js [hunk 6]
```diff
@@ -46,6 +46,23 @@ export function createFile(fileInformation, parentFolder, name) {
   parentFolder.children.push(newFile);
 }
 
+/**
+ * Update a file.
```
[the remark this answers is not in the document]
Comment (k-y0u) [reply src/composables/FileExplorer.js:50]: the update updates, it's all you need to know O_O

File: src/composables/FileExplorer.js [hunk 7]
```diff
@@ -46,6 +46,23 @@ export function createFile(fileInformation, parentFolder, name) {
...
+ * @param {number} index - Index of the part to check in the path.
+ */
+export function updateFile(folder, fileStatus, index = 0) {
+  const splitPath = fileStatus.path.split('/');
+  const item = folder.children.find((child) => child.label === splitPath[index]);
+
+  if (splitPath.length === index + 1) {
+    item.information = fileStatus;
+  } else {
+    updateFile(item, fileStatus, index + 1);
```
Comment (Zorin95670) [inline src/composables/FileExplorer.js:62]: status of depend of all files in it not only one ?

File: src/components/editor/MonacoEditor.vue [hunk 8]
```diff
@@ -58,7 +59,12 @@ async function updateFile() {
   };
 
   await writeProjectFile(props.projectName, file);
-  FileEvent.UpdateEditorContentEvent.next(file.id);
+  const [fileStatus] = await getStatus(
```
Comment (Zorin95670) [inline src/components/editor/MonacoEditor.vue:62]: break line before

File: src/components/editor/MonacoEditor.vue (hunk 5 above)
Comment (Zorin95670) [reply src/components/editor/MonacoEditor.vue:113]: Externalize function inside `onDidChangeModelContent` and add documentation on it.

File: src/composables/FileExplorer.js (hunk 7 above)
Comment (Zorin95670) [reply src/composables/FileExplorer.js:62]: see with @k-y0u

File: src/components/editor/MonacoEditor.vue (hunk 8 above)
Comment (k-y0u) [reply src/components/editor/MonacoEditor.vue:63]: done

File: src/components/editor/MonacoEditor.vue (hunk 5 above)
Comment (k-y0u) [reply src/components/editor/MonacoEditor.vue:133]: done

**Review by aboubakar14 — APPROVED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**